raiden/network/transport/matrix/transport.py

```diff
@@ -1246,97 +1250,72 @@ def _handle_sync_messages(self, sync_messages: MatrixSyncMessages) -> bool:
 
         assert self._raiden_service is not None, "_raiden_service not set"
 
-        all_messages: List[Message] = list()
+        raiden_messages: List[Message] = list()
+        call_messages: List[Tuple[Address, MatrixMessage]] = list()
+
         for room, room_messages in sync_messages:
-            for text in room_messages:
-                all_messages.extend(self._handle_text(room, text))
+            raiden_room_messages, call_room_messages = self._validate_matrix_messages(
+                room, room_messages
+            )
+            raiden_messages.extend(raiden_room_messages)
+            call_messages.extend(call_room_messages)
 
-        self._process_messages(all_messages)
-        return len(all_messages) > 0
+        self._process_messages(raiden_messages)
+        self._process_call_messages(call_messages)
+        return len(raiden_messages) > 0 or len(call_messages) > 0
 
     def _handle_web_rtc_messages(self, message_data: str, partner_address: Address) -> None:
         if not self._stop_event.is_set():
             messages = validate_and_parse_message(message_data, partner_address)
             self._process_messages(messages)
 
-    def _handle_call_messages(self, sync_messages: MatrixSyncMessages) -> None:
+    def _process_call_messages(self, call_messages: List[Tuple[Address, MatrixMessage]]) -> None:
         """
         This function handles incoming signalling messages (in matrix called 'call' events).
         In Raiden 'm.room.message' events are used as the communication format.
         Main function is to forward it to the aiortc library to establish connections.
         Messages contain sdp messages to follow the ROAP (RTC Offer Answer Protocol).
         Args:
-            sync_messages: List of call events received by a partner (either offer or answer)
+            call_messages: List of signalling messages
         """
         assert self._raiden_service is not None, "_raiden_service not set"
 
-        for room, call_events in sync_messages:
-            for call_event in call_events:
+        for partner_address, call_message in call_messages:
 
-                # Ignore our own messages. We can use address_from_userid in this specific case
-                # since it is a negative check. This is used to avoid messages from same address
-                # but different user (due to roaming)
-                sender_id = call_event["sender"]
-                if self._raiden_service.address == address_from_userid(sender_id):
-                    continue
+            content = json.loads(call_message["content"]["body"])
```

Comment:
Unless I'm misreading this `loads` could fail for incorrectly formatted messages.